lib/rules/at-rule-prelude-no-invalid/index.cjs

```diff
@@ -0,0 +1,78 @@
+// NOTICE: This file is generated by Rollup. To modify it,
+// please instead edit the ESM counterpart and rebuild with Rollup (npm run build).
+'use strict';
+
+const cssTree = require('css-tree');
+const validateTypes = require('../../utils/validateTypes.cjs');
+const isStandardSyntaxAtRule = require('../../utils/isStandardSyntaxAtRule.cjs');
+const optionsMatches = require('../../utils/optionsMatches.cjs');
+const report = require('../../utils/report.cjs');
+const ruleMessages = require('../../utils/ruleMessages.cjs');
+const validateOptions = require('../../utils/validateOptions.cjs');
+
+const ruleName = 'at-rule-prelude-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (atRule, prelude) => `Unexpected invalid prelude "${prelude}" for at-rule "${atRule}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/at-rule-prelude-no-invalid',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreAtRules: [validateTypes.isString, validateTypes.isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules((atRule) => {
+			if (!isStandardSyntaxAtRule(atRule)) return;
+
+			if (optionsMatches(secondaryOptions, 'ignoreAtRules', atRule.name)) return;
+
+			const { name, params } = atRule;
+
+			const { error } = cssTree.lexer.matchAtrulePrelude(name, params);
+
+			if (!error) return;
+
+			if (error.name !== 'SyntaxMatchError') return;
+
+			const atName = `@${name}`;
+
+			const index = atName.length + 1;
```

Comment:
question

This assumes there's always exactly one space.
e.g. [`@page:blank:first {}`](https://github.com/stylelint/stylelint/pull/8176#discussion_r1874248370)

Could you are add a test for `@property  foo {}`?
i.e. 2 spaces

Reply:
Good catch!
I made the fixes based on this suggestions.
https://github.com/stylelint/stylelint/pull/8268#discussion_r1901742644

I also added tests.
[Apply Mouvedia's suggestions](https://github.com/stylelint/stylelint/pull/8268/commits/9e6a1d0cf11150fa83f489945aaf40b2e2a21a31)